src/backend/expungeservice/expunger/expunger.py

```diff
@@ -123,4 +126,46 @@ def _class_b_felonies(charges):
         for charge in charges:
             if isinstance(charge, FelonyClassB):
                 class_b_felonies.append(charge)
-        return class_b_felonies
\ No newline at end of file
+        return class_b_felonies
+
+    @staticmethod
+    def _mark_missing_or_unknown_dispositions_as_record_errors(record):
+        cases_with_missing_disposition : List[str] = []
+        cases_with_unknown_disposition : List[str] = []
+        for charge in record.charges:
+            if not charge.skip_analysis():
+                case_number = charge.case()().case_number
+                if not charge.disposition:
+                    if not case_number in cases_with_missing_disposition:
+                        cases_with_missing_disposition.append(case_number)
+                elif charge.disposition.status == DispositionStatus.UNKNOWN:
+                    if not case_number in cases_with_unknown_disposition:
+                        cases_with_unknown_disposition.append(case_number)
+            message_shared_content = (
+                "This is likely an error in the OECI database. " +
+                "Time analysis is ignoring these charges and may be inaccurate for other charges.")
+            if cases_with_missing_disposition:
+                if len(cases_with_missing_disposition) == 1:
+                    missing_dispo_error_message = (
+                        f"Case {cases_with_missing_disposition[0]} has a charge with a missing disposition. " +
+                        message_shared_content)
+                else:
+                    cases_list_string = ", ".join(cases_with_missing_disposition)
+                    missing_dispo_error_message = (
+                        "The following cases have charges with a missing disposition. " +
+                        message_shared_content +
+                        f"\nCase numbers: {cases_list_string}")
+                record.errors.append(missing_dispo_error_message)
+
+            if cases_with_unknown_disposition:
+                if len(cases_with_unknown_disposition) == 1:
+                    unknown_dispo_error_message = (
+                        f"Case {cases_with_unknown_disposition[0]} has a charge with an unrecognized disposition. " +
+                        message_shared_content)
+                else:
+                    cases_list_string = ", ".join(cases_with_unknown_disposition)
+                    unknown_dispo_error_message = (
+                        "The following cases have charges with an unrecognized disposition." +
+                        message_shared_content +
+                        f"\nCase numbers: {cases_list_string}")
+                record.errors.append(unknown_dispo_error_message)
```

Comment:
I would prefer to pass in `self.charges`, return a list of record errors (`List[str]`), and then `+=` the result to record.errors in the `run` method. We want to keep all private functions in this class side-effect free.

Reply:
Will do. I thought about that and decided to keep the `run` method looking clean, but I think you're right that the functional style is more important here.